frontend/pages/hosts/details/HostDetailsPage/components/HostDetailsBanners/HostDetailsBanners.tsx

```diff
@@ -21,26 +23,45 @@ const HostDetailsBanners = ({
   mdmName,
   diskEncryptionStatus,
 }: IHostDetailsBannersProps) => {
-  const { config, isSandboxMode, isPremiumTier } = useContext(AppContext);
+  const { config, isPremiumTier, apnsExpiry, abmExpiry } = useContext(
+    AppContext
+  );
 
-  // checks to see if the ABM message is being shown in a parent component.
-  // We want this to be the only banner shown to the user so we need to know
-  // if it's already being shown so we can suppress other banners.
+  // Checks to see if an app-wide banner is being shown (the ABM terms, ABM expiry,
+  // or APNs expiry banner) in a parent component. App-wide banners found in parent
+  // component take priority over host details page-level banners.
   const isAppleBmTermsExpired = config?.mdm?.apple_bm_terms_expired;
-  const showingAppleABMBanner =
-    (isAppleBmTermsExpired && isPremiumTier && !isSandboxMode) ?? false;
+  const isApplePnsExpired = hasLicenseExpired(apnsExpiry || "");
+  const willApplePnsExpireIn30Days = willExpireWithinXDays(
+    apnsExpiry || "",
+    30
+  );
+  const isAppleBmExpired = hasLicenseExpired(abmExpiry || "");
+  const willAppleBmExpireIn30Days = willExpireWithinXDays(abmExpiry || "", 30);
+  const isFleetLicenseExpired = hasLicenseExpired(
+    config?.license.expiration || ""
+  );
+
+  const showingAppWideBanner =
```

Comment:
I wonder if a more directly linked way to check this would be more bug-proof? Not a deal breaker.

Reply:
Yeah, I just built on what's already in there. Definitely consider a refactor if more time was allotted.